Add moose-opt (combined-opt) INL HPC instructions

milljm opened the pull request:
Add inl_hpc_prebuilt_moose.md. Detailing instructions on how to use `moose-opt` and how to view the results.

Fix grammar in clone_moose.md

Closes #23351

Commit:
Add moose-opt (combined-opt) INL HPC instructions

Add inl_hpc_prebuilt_moose.md. Detailing instructions on how to use
`moose-opt` and how to view the results.

Fix grammar in clone_moose.md

Closes #23351

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1334319/) on 184ac1a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23355/site)

This comment will be updated on new commits.

GiudGiud commented:
Awesome! This will facilitate trainings like the one we gave at NIST.

milljm commented:
There are more changes incoming.

Commit:
Update Manual Installation instructions

Fix some grammar for INL MOOSE Binary instructions.

Add peacock package install instructions.

Correct a bunch of conda/mamba syntax.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1337355/) on 41acc7a wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>982f25</th>
      <th colspan=4>#23355 41acc7</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.70%</td>
      <td>84.70%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>82862</td>
      <td>82865</td>
      <td>+3</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14966</td>
      <td>14963</td>
      <td>-3</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23355/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23355/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/23355/coverage/xfem)

</details>



This comment will be updated on new commits.

cticenhour commented on modules/doc/content/getting_started/installation/clone_moose.md:
```diff
@@ -1,8 +1,10 @@
 
-MOOSE is hosted on [GitHub](https://github.com/idaholab/moose) and should be cloned directly from there using [git](https://git-scm.com/). We recommend creating a directory named projects to put all of your MOOSE related work.
+MOOSE is hosted on [GitHub](https://github.com/idaholab/moose) and should be cloned directly from
+there using [git](https://git-scm.com/). We recommend creating a directory named projects to put all
```
```suggestion
there using [git](https://git-scm.com/). We recommend creating a directory named "projects" to put all
```

cticenhour commented on modules/doc/content/getting_started/installation/index.md:
```diff
@@ -1,10 +1,12 @@
 
-To install the MOOSE Framework, click the link below that corresponds to your operating system/platform and follow the instructions:
+To install the MOOSE Framework, click the link below that corresponds to your operating
+system/platform or your desired method and follow the instructions:
```
```suggestion
system, platform, or your desired method and follow the instructions:
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
```diff
@@ -0,0 +1,86 @@
+# INL HPC Pre-Built MOOSE
+
+While operating on one of our [!ac](INL) [!ac](HPC) clusters, there exists the option of using pre
+built versions of MOOSE. To request access to these clusters, please follow the instructions on
```
```suggestion
While operating on one of the [!ac](INL) [!ac](HPC) clusters, there exists the option of using pre-built 
versions of MOOSE. To request access to these clusters, please follow the instructions on
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
````diff
@@ -0,0 +1,86 @@
+```
+
+!alert warning
+If you receive an error about modules not known, please make sure you are logged into either
````
```suggestion
If you receive an error about modules not being known, please make sure you are logged into either
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
```diff
@@ -0,0 +1,86 @@
+Sawtooth), the amount of time you believe you need, and then click `Launch`.
+
+It may take some time before your 'Visualization Job' becomes available. When it does, simply click
+on it, and you will be presented within your web browser, your GUI Desktop. From here, you can open
```
```suggestion
on it, and you will be presented a [!ac](GUI) desktop within your web browser. From here, you can open
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
```diff
@@ -0,0 +1,86 @@
+on it, and you will be presented within your web browser, your GUI Desktop. From here, you can open
+visualization applications (such as Paraview), and open your results file.
+
+To use Paraview, open a terminal by: Clicking `Applications` at the top left, then click
```
```suggestion
To use Paraview, open a terminal by clicking `Applications` at the top left, then click
```

cticenhour commented on modules/doc/content/getting_started/installation/install_conda_moose.md:
````diff
@@ -1,13 +1,6 @@
-```
-
-Next, create a unique conda environment for moose, named `moose`, and install the moose dependency
+Create a unique conda environment for moose, named `moose`, and install the moose dependency
````
```suggestion
Create a unique conda environment for [!ac](MOOSE), named `moose`, and install the MOOSE dependency
```

cticenhour commented on modules/doc/content/getting_started/installation/install_miniconda.md:
````diff
@@ -24,17 +25,33 @@ guide relies on the utilization of `mamba`, an optimized package manager for Con
 ```
 
-Configure Conda to work with our INL public channel:
+Now that we can execute `mamba`, initailize it and then exit the terminal:
````
```suggestion
Now that we can execute `mamba`, initialize it and then exit the terminal:
```

cticenhour commented on modules/doc/content/getting_started/installation/manual_installation_llvm.md:
````diff
@@ -34,7 +34,25 @@ export LD_LIBRARY_PATH=$PACKAGES_DIR/llvm-__LLVM__/lib:$LD_LIBRARY_PATH
 !include manual_mpich_llvm.md
 
-!include getting_started/installation/manual_miniconda.md
+## Python (possibly optional)
+
+Building MOOSE requires having Python headers (python-devel) and the Python package: `packaging`
+available. If you know you have this requirement satisfied, and you do not plan on also using
+Peacock (next section) you can skip this section.
+
+These requirements are easily installed via Conda.
+
+!include getting_started/installation/install_miniconda.md
+
+Next, install the necessary `packaging` module:
+
+```bash
+mamba install packaging
+```
+
+## Peacock (optional)
+
+!include installation/install_peacock.md
````
Since this is used twice - is it better to put this content in a single file we include? Even if it results in some patching pain.

cticenhour commented on modules/doc/content/help/faq/conda_issues.md:
```diff
@@ -1,18 +1,21 @@
 
-Conda issues can be the root cause for just about any issue on this page. Scroll through this section, for what may look familiar, and follow those instructions:
+Conda issues can be the root cause for just about any issue on this page. Scroll through this
+section, for what may look familiar, and follow those instructions:
```
```suggestion
section for what may look familiar, and follow those instructions:
```

cticenhour commented on modules/doc/content/help/faq/conda_issues.md:
```diff
@@ -28,56 +31,54 @@ Conda issues can be the root cause for just about any issue on this page. Scroll
+  installed (perhaps with sudo rights, or as another user). You should look into removing this
+  installation of conda, and starting over with our
+  [Getting Started](getting_started/installation/conda.md) instructions. Failures of this nature can
+  also mean your conda resource file (~/.condarc) is in bad shape. We have no way of diagnosing this
```
```suggestion
  also mean your conda resource file (`~/.condarc`) is in bad shape. We have no way of diagnosing this
```

cticenhour commented on modules/doc/content/help/faq/conda_issues.md:
```diff
@@ -86,38 +87,29 @@ Conda issues can be the root cause for just about any issue on this page. Scroll
 - #### Your issue not listed
 
-  The quick fix-attempt, is to re-install the moose-packages:
+  The quick fix-attempt, is to delete the faulty environment and re-install it again:
```
```suggestion
  The quick fix-attempt, is to delete the faulty environment and re-install it:
```

Commit:
Fix grammar/spelling

Fix grammar and spelling errors.
Consolidate the optional Python/Peacock instructions.

cticenhour reviewed (approved):
There doesn't appear to be a Griffin patch (I can make one shortly). 

Patches to RELAP-7, Sockeye, and Mastodon ready for a submodule update on merge.

cticenhour commented:
Griffin patch is ready for a submodule update: https://github.inl.gov/ncrc/griffin/pull/1056

I think we're ready to go here in the morning.

milljm commented:
> Griffin patch is ready for a submodule update: https://github.inl.gov/ncrc/griffin/pull/1056
> 
> I think we're ready to go here in the morning.

I had one created... this one: https://github.inl.gov/ncrc/griffin/issues/1054
